### PR TITLE
Add Uniform distribution

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**
split from #4678 .

### CODE REVIEW

**Comment by toslunar** [comment]
How should we deal with the difference of arguments between `numpy.random` and `scipy.stats`?

- `numpy.random.uniform` takes `low` and `high`
- `scipy.stats.uniform` takes `loc` and `scale`.

(`low == loc`, `high == loc + scale`) 

It seems the safest way, at the moment, to make them keyword-only arguments.

**Comment by YoshikawaMasashi** [comment]
I think you are right.
I will implement this feature

File: tests/chainer_tests/distributions_tests/test_kldivergence.py [hunk 1]
```diff
@@ -125,5 +134,15 @@ def test_multivariatenormal_multivariatenormal_gpu(self):
...
+    def test_uniform_uniform_cpu(self):
+        dist1 = self.make_uniform_dist()
+        dist2 = self.make_uniform_dist()
+        self.check_kl(dist1, dist2)
+
+    @attr.gpu
+    def test_uniform_uniform_gpu(self):
+        dist1 = self.make_uniform_dist(True)
+        dist2 = self.make_uniform_dist(True)
+        self.check_kl(dist1, dist2)
```
Comment (toslunar) [inline tests/chainer_tests/distributions_tests/test_kldivergence.py]: Fix style: two empty lines are needed below.

File: tests/chainer_tests/distributions_tests/test_uniform.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        self.scipy_dist = stats.uniform
+
+        self.test_targets = set([
+            "batch_shape", "cdf", "entropy", "event_shape", "icdf", "log_prob",
+            "mean", "sample", "stddev", "support", "variance"])
+
+        low = numpy.random.uniform(-10, 0, self.shape).astype(numpy.float32)
+        high = numpy.random.uniform(
+            low, low + 10, self.shape).astype(numpy.float32)
+        self.params = {"low": low, "high": high}
```
Comment (toslunar) [inline tests/chainer_tests/distributions_tests/test_uniform.py]: Could you test "loc" and "scale", too?

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Pull request merged**